src/Microsoft.ML.Transforms/Text/TextNormalizerTransform.cs

```diff
@@ -62,28 +60,30 @@ public sealed class Arguments
             public Column[] Column;
 
             [Argument(ArgumentType.AtMostOnce, HelpText = "Casing text using the rules of the invariant culture.", ShortName = "case", SortOrder = 1)]
-            public CaseNormalizationMode TextCase = CaseNormalizationMode.Lower;
+            public TextNormalizerEstimator.CaseNormalizationMode TextCase = TextNormalizerEstimator.Defaults.TextCase;
 
             [Argument(ArgumentType.AtMostOnce, HelpText = "Whether to keep diacritical marks or remove them.",
                 ShortName = "diac", SortOrder = 1)]
-            public bool KeepDiacritics = false;
+            public bool KeepDiacritics = TextNormalizerEstimator.Defaults.KeepDiacritics;
 
             [Argument(ArgumentType.AtMostOnce, HelpText = "Whether to keep punctuation marks or remove them.", ShortName = "punc", SortOrder = 2)]
-            public bool KeepPunctuations = true;
+            public bool KeepPunctuations = TextNormalizerEstimator.Defaults.KeepPunctuations;
 
             [Argument(ArgumentType.AtMostOnce, HelpText = "Whether to keep numbers or remove them.", ShortName = "num", SortOrder = 2)]
-            public bool KeepNumbers = true;
+            public bool KeepNumbers = TextNormalizerEstimator.Defaults.KeepNumbers;
         }
 
         internal const string Summary = "A text normalization transform that allows normalizing text case, removing diacritical marks, punctuation marks and/or numbers." +
             " The transform operates on text input as well as vector of tokens/text (vector of ReadOnlyMemory).";
 
-        public const string LoaderSignature = "TextNormalizerTransform";
+        public const string LoaderSignature = nameof(TextNormalizerTransform);
+
         private static VersionInfo GetVersionInfo()
         {
             return new VersionInfo(
                 modelSignature: "TEXTNORM",
-                verWrittenCur: 0x00010001, // Initial
+                //verWrittenCur: 0x00010001, // Initial
+                verWrittenCur: 0x00010002, // Params for each column.
```

Comment:
>verWrittenCur: 0x00010002, // Params for each column. [](start = 16, length = 53)

@Zruty0, I'm actually not sure how benefitial and usefull to provide options for each column rather than provide set of columns and options for all of them. Would be nice if you can comment.

Reply:
This would unblock certain pigsty optimizations, where you can bundle multiple calls to text normalizer (with different params) into one estimator. But since this is not incurring a data pass, I'm ok with not having a per-column option.

---
In reply to: [226042341](https://github.com/dotnet/machinelearning/pull/1276#discussion_r226042341) [](ancestors = 226042341)